packages/financial-templates-lib/src/price-feed/BalancerPriceFeed.js

```diff
@@ -27,8 +30,28 @@ class BalancerPriceFeed extends PriceFeedInterface {
     // Add a callback to get price, error can be thrown from web3 disconection or maybe something else
     // which affects the update call.
     this.priceHistory = PriceHistory(async number => {
-      return this.toBN(await this.contract.methods.getSpotPriceSansFee(this.tokenIn, this.tokenOut).call(number));
+      try {
+        return this.toBN(await this.contract.methods.getSpotPriceSansFee(this.tokenIn, this.tokenOut).call(number));
```

Comment:
I need to update this method to compute the TWAP instead of spot price, like the Uniswap price feed.